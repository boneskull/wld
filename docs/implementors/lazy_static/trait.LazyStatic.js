(function() {var implementors = {};
implementors["wld"] = [{"text":"impl LazyStatic for BLOCKTYPE_COLOR_MAP","synthetic":false,"types":[]},{"text":"impl LazyStatic for WALLTYPE_COLOR_MAP","synthetic":false,"types":[]},{"text":"impl LazyStatic for PAINT_COLOR_MAP","synthetic":false,"types":[]}];
if (window.register_implementors) {window.register_implementors(implementors);} else {window.pending_implementors = implementors;}})()