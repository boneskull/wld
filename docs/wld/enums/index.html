<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `enums` mod in crate `wld`."><meta name="keywords" content="rust, rustlang, rust-lang, enums"><title>wld::enums - Rust</title><link rel="stylesheet" type="text/css" href="../../normalize.css"><link rel="stylesheet" type="text/css" href="../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../../ayu.css" disabled ><script src="../../storage.js"></script><noscript><link rel="stylesheet" href="../../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../../favicon.svg">
<link rel="alternate icon" type="image/png" href="../../favicon-16x16.png">
<link rel="alternate icon" type="image/png" href="../../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../../wld/index.html'><div class='logo-container rust-logo'><img src='../../rust-logo.png' alt='logo'></div></a><p class="location">Module enums</p><div class="sidebar-elems"><div class="block items"><ul><li><a href="#enums">Enums</a></li></ul></div><p class="location"><a href="../index.html">wld</a></p><script>window.sidebarCurrent = {name: "enums", ty: "mod", relpath: "../"};</script><script defer src="../sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!"><img src="../../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices"></div></div><script src="../../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><span class="help-button">?</span>
                <a id="settings-menu" href="../../settings.html"><img src="../../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../../src/wld/enums.rs.html#1-26" title="goto source code">[src]</a></span><span class="in-band">Module <a href="../index.html">wld</a>::<wbr><a class="mod" href="">enums</a></span></h1><h2 id="enums" class="section-header"><a href="#enums">Enums</a></h2>
<table><tr class="module-item"><td><a class="enum" href="enum.AnglerQuestFish.html" title="wld::enums::AnglerQuestFish enum">AnglerQuestFish</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.BlockShape.html" title="wld::enums::BlockShape enum">BlockShape</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.BlockType.html" title="wld::enums::BlockType enum">BlockType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.EntityType.html" title="wld::enums::EntityType enum">EntityType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.EvilType.html" title="wld::enums::EvilType enum">EvilType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.HardmodeOreType.html" title="wld::enums::HardmodeOreType enum">HardmodeOreType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.InvasionType.html" title="wld::enums::InvasionType enum">InvasionType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.ItemType.html" title="wld::enums::ItemType enum">ItemType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.LiquidType.html" title="wld::enums::LiquidType enum">LiquidType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.RLEType.html" title="wld::enums::RLEType enum">RLEType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.TBool.html" title="wld::enums::TBool enum">TBool</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.TileEntityType.html" title="wld::enums::TileEntityType enum">TileEntityType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.WallType.html" title="wld::enums::WallType enum">WallType</a></td><td class="docblock-short"><p>A list of vanilla Walls by internal name, mapped to their IDs.</p>
</td></tr></table></section><section id="search" class="content hidden"></section><section class="footer"></section><script>window.rootPath = "../../";window.currentCrate = "wld";</script><script src="../../main.js"></script><script defer src="../../search-index.js"></script></body></html>