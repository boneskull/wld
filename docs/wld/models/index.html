<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `models` mod in crate `wld`."><meta name="keywords" content="rust, rustlang, rust-lang, models"><title>wld::models - Rust</title><link rel="stylesheet" type="text/css" href="../../normalize.css"><link rel="stylesheet" type="text/css" href="../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../../ayu.css" disabled ><script src="../../storage.js"></script><noscript><link rel="stylesheet" href="../../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../../favicon.svg">
<link rel="alternate icon" type="image/png" href="../../favicon-16x16.png">
<link rel="alternate icon" type="image/png" href="../../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../../wld/index.html'><div class='logo-container rust-logo'><img src='../../rust-logo.png' alt='logo'></div></a><p class="location">Module models</p><div class="sidebar-elems"><div class="block items"><ul><li><a href="#structs">Structs</a></li><li><a href="#types">Type Definitions</a></li></ul></div><p class="location"><a href="../index.html">wld</a></p><script>window.sidebarCurrent = {name: "models", ty: "mod", relpath: "../"};</script><script defer src="../sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!"><img src="../../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices"></div></div><script src="../../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><span class="help-button">?</span>
                <a id="settings-menu" href="../../settings.html"><img src="../../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../../src/wld/models.rs.html#1-35" title="goto source code">[src]</a></span><span class="in-band">Module <a href="../index.html">wld</a>::<wbr><a class="mod" href="">models</a></span></h1><h2 id="structs" class="section-header"><a href="#structs">Structs</a></h2>
<table><tr class="module-item"><td><a class="struct" href="struct.AnglerQuestStatus.html" title="wld::models::AnglerQuestStatus struct">AnglerQuestStatus</a></td><td class="docblock-short"><p>A bucket for a few angler-related things.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Backgrounds.html" title="wld::models::Backgrounds struct">Backgrounds</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Block.html" title="wld::models::Block struct">Block</a></td><td class="docblock-short"><p>Represents a block.  A <a href="../../wld/models/struct.Tile.html"><code>Tile</code></a> optionally &quot;has&quot; a <a href="../../wld/models/struct.Block.html" title="Block"><code>Block</code></a>.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.BlockCtx.html" title="wld::models::BlockCtx struct">BlockCtx</a></td><td class="docblock-short"><p>Contains some metadata from the <a href="../../wld/models/struct.Tile.html"><code>Tile</code></a> instance to which a
<a href="../../wld/models/struct.Block.html" title="Block"><code>Block</code></a> belongs.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.BossesSlain.html" title="wld::models::BossesSlain struct">BossesSlain</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.BossesSlain2.html" title="wld::models::BossesSlain2 struct">BossesSlain2</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Bounds.html" title="wld::models::Bounds struct">Bounds</a></td><td class="docblock-short"><p>Represents the size of a Terraria map, as a rectangle.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Chest.html" title="wld::models::Chest struct">Chest</a></td><td class="docblock-short"><p>Represents a chest.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Chests.html" title="wld::models::Chests struct">Chests</a></td><td class="docblock-short"><p>Represents all <a href="../../wld/models/struct.Chest.html" title="Chest"><code>Chest</code></a>s in a Terraria world.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Clouds.html" title="wld::models::Clouds struct">Clouds</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.EventsCompleted.html" title="wld::models::EventsCompleted struct">EventsCompleted</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.ExtendedTileAttributes.html" title="wld::models::ExtendedTileAttributes struct">ExtendedTileAttributes</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Footer.html" title="wld::models::Footer struct">Footer</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.GeneratorInfo.html" title="wld::models::GeneratorInfo struct">GeneratorInfo</a></td><td class="docblock-short"><p>Information about the world seed.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.HardmodeOre.html" title="wld::models::HardmodeOre struct">HardmodeOre</a></td><td class="docblock-short"><p>Represents one of three types of ore available in hardmode.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Header.html" title="wld::models::Header struct">Header</a></td><td class="docblock-short"><p>World file header.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.House.html" title="wld::models::House struct">House</a></td><td class="docblock-short"><p>Represents NPC housing.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.HouseVec.html" title="wld::models::HouseVec struct">HouseVec</a></td><td class="docblock-short"><p>A variable-length list of <a href="../../wld/models/struct.House.html" title="House"><code>House</code></a>s.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.InvasionStatus.html" title="wld::models::InvasionStatus struct">InvasionStatus</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.ItemStack.html" title="wld::models::ItemStack struct">ItemStack</a></td><td class="docblock-short"><p>Represents a stack of items in a <a href="../../wld/models/struct.Chest.html" title="Chest"><code>Chest</code></a>.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.ItemStackVec.html" title="wld::models::ItemStackVec struct">ItemStackVec</a></td><td class="docblock-short"><p>A list of <a href="../../wld/models/struct.ItemStack.html" title="ItemStack"><code>ItemStack</code></a>s, which make up the contents of a <a href="../../wld/models/struct.Chest.html" title="Chest"><code>Chest</code></a>.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Liquid.html" title="wld::models::Liquid struct">Liquid</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.LogicSensor.html" title="wld::models::LogicSensor struct">LogicSensor</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Mob.html" title="wld::models::Mob struct">Mob</a></td><td class="docblock-short"><p>Represents a &quot;mob&quot;; a non-friendly NPC entity.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.MobKills.html" title="wld::models::MobKills struct">MobKills</a></td><td class="docblock-short"><p>A list of kill counts corresponding to mobs.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.MobVec.html" title="wld::models::MobVec struct">MobVec</a></td><td class="docblock-short"><p>List of <a href="../../wld/models/struct.Mob.html" title="Mob"><code>Mob</code></a>s; fancy wrapper around a <a href="https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html" title="Vec"><code>Vec</code></a>.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.NPC.html" title="wld::models::NPC struct">NPC</a></td><td class="docblock-short"><p>Represents a friendly NPC.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.NPCVec.html" title="wld::models::NPCVec struct">NPCVec</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Offsets.html" title="wld::models::Offsets struct">Offsets</a></td><td class="docblock-short"><p>Byte offsets in a <code>.wld</code> file corresponding to certain data points.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.OldOnesArmyStatus.html" title="wld::models::OldOnesArmyStatus struct">OldOnesArmyStatus</a></td><td class="docblock-short"><p>Represents the unlocked tiers of the Old One's Army event.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.PartyStatus.html" title="wld::models::PartyStatus struct">PartyStatus</a></td><td class="docblock-short"><p>Various bits of party-related information.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.PartyingNPCs.html" title="wld::models::PartyingNPCs struct">PartyingNPCs</a></td><td class="docblock-short"><p>A list of partying NPCs.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.PillarStatus.html" title="wld::models::PillarStatus struct">PillarStatus</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Position.html" title="wld::models::Position struct">Position</a></td><td class="docblock-short"><p>Represents a position (or point) in terms of cartesian coordinates.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.PressurePlates.html" title="wld::models::PressurePlates struct">PressurePlates</a></td><td class="docblock-short"><p>Represents all <a href="../../wld/models/type.PressurePlate.html" title="PressurePlate"><code>PressurePlate</code></a>s in the game world.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Properties.html" title="wld::models::Properties struct">Properties</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.QuadrantStyle.html" title="wld::models::QuadrantStyle struct">QuadrantStyle</a></td><td class="docblock-short"><p>Represents styles which differ by &quot;quandrant&quot; of the map (by X-axis).</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.RainStatus.html" title="wld::models::RainStatus struct">RainStatus</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.RunLength.html" title="wld::models::RunLength struct">RunLength</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.SandstormStatus.html" title="wld::models::SandstormStatus struct">SandstormStatus</a></td><td class="docblock-short"><p>Bucket for info about sandstorm events.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.SavedNPCs.html" title="wld::models::SavedNPCs struct">SavedNPCs</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.ShadowOrbStatus.html" title="wld::models::ShadowOrbStatus struct">ShadowOrbStatus</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Sign.html" title="wld::models::Sign struct">Sign</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Signs.html" title="wld::models::Signs struct">Signs</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Status.html" title="wld::models::Status struct">Status</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TBitVec.html" title="wld::models::TBitVec struct">TBitVec</a></td><td class="docblock-short"><p>A fixed-length vector of bits; a fancy wrapper around <a href="bitvec::vec::BitVec"><code>BitVec</code></a>.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.TString.html" title="wld::models::TString struct">TString</a></td><td class="docblock-short"><p>Represents a string as per the data format; a fancy wrapper around <a href="https://doc.rust-lang.org/nightly/alloc/string/struct.String.html" title="String"><code>String</code></a>.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.TUuid.html" title="wld::models::TUuid struct">TUuid</a></td><td class="docblock-short"><p>A UUID.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Tile.html" title="wld::models::Tile struct">Tile</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TileAttributes.html" title="wld::models::TileAttributes struct">TileAttributes</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TileCtx.html" title="wld::models::TileCtx struct">TileCtx</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TileEntities.html" title="wld::models::TileEntities struct">TileEntities</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TileEntity.html" title="wld::models::TileEntity struct">TileEntity</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TileHeader.html" title="wld::models::TileHeader struct">TileHeader</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TileMatrix.html" title="wld::models::TileMatrix struct">TileMatrix</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TileVec.html" title="wld::models::TileVec struct">TileVec</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.TownManager.html" title="wld::models::TownManager struct">TownManager</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Uuid.html" title="wld::models::Uuid struct">Uuid</a></td><td class="docblock-short"><p>A Universally Unique Identifier (UUID).</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.VariableTBitVec.html" title="wld::models::VariableTBitVec struct">VariableTBitVec</a></td><td class="docblock-short"><p>A variable-length vector of bits; a fancy wrapper around <a href="bitvec::vec::BitVec"><code>BitVec</code></a>.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.Wall.html" title="wld::models::Wall struct">Wall</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.Wiring.html" title="wld::models::Wiring struct">Wiring</a></td><td class="docblock-short"><p>Represents wires on a <a href="../../wld/models/struct.Tile.html" title="Tile"><code>Tile</code></a>.  A <code>Tile</code> can have all of these, none of
these, or any in between.
For more information, see <a href="https://terraria.gamepedia.com/Wire">Wire</a> on
the <a href="https://terraria.gamepedia.com">Official Terraria Wiki</a>.</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.World.html" title="wld::models::World struct">World</a></td><td class="docblock-short"><p>Top-level representation of a Terraria world file (<code>.wld</code>).</p>
</td></tr><tr class="module-item"><td><a class="struct" href="struct.WorldCtx.html" title="wld::models::WorldCtx struct">WorldCtx</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="struct" href="struct.WorldStyle.html" title="wld::models::WorldStyle struct">WorldStyle</a></td><td class="docblock-short"><p>A bucket in which some style information lives.</p>
</td></tr></table><h2 id="types" class="section-header"><a href="#types">Type Definitions</a></h2>
<table><tr class="module-item"><td><a class="type" href="type.HellStyle.html" title="wld::models::HellStyle type">HellStyle</a></td><td class="docblock-short"><p>Represents the appearance of the hell biome's background.</p>
</td></tr><tr class="module-item"><td><a class="type" href="type.MoonStyle.html" title="wld::models::MoonStyle type">MoonStyle</a></td><td class="docblock-short"><p>Represents the appearance of the moon.</p>
</td></tr><tr class="module-item"><td><a class="type" href="type.PressurePlate.html" title="wld::models::PressurePlate type">PressurePlate</a></td><td class="docblock-short"><p>Represents a pressure plate.</p>
</td></tr><tr class="module-item"><td><a class="type" href="type.UndergroundJungleStyle.html" title="wld::models::UndergroundJungleStyle type">UndergroundJungleStyle</a></td><td class="docblock-short"><p>Represents the appearance of the underground jungle biome's background.</p>
</td></tr><tr class="module-item"><td><a class="type" href="type.UndergroundSnowStyle.html" title="wld::models::UndergroundSnowStyle type">UndergroundSnowStyle</a></td><td class="docblock-short"><p>Represents the appearance of the underground snow biome's background.</p>
</td></tr></table></section><section id="search" class="content hidden"></section><section class="footer"></section><script>window.rootPath = "../../";window.currentCrate = "wld";</script><script src="../../main.js"></script><script defer src="../../search-index.js"></script></body></html>