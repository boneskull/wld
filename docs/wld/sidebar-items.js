initSidebarItems({"fn":[["enable_logger","Enable logging for this crate given [`log::LevelFilter`] `level`."],["parse_world","Instantiate a [`World`] from a slice of bytes."]],"mod":[["constants",""],["enums",""],["models",""]]});