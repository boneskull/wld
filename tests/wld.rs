use insta::assert_debug_snapshot;
use lazy_static::lazy_static;
use std::fs::read;
use wld::{model::world::World, parse_world};

lazy_static! {
  // paths are relative to root, I guess?
  pub static ref WORLD: std::vec::Vec<u8> =
    read("tests/fixtures/Foon.wld").expect("Unable to read file");
  pub static ref PARSED_WORLD: World = parse_world(&WORLD).unwrap();
}

#[test]
fn test_parse_header() {
  assert_debug_snapshot!(PARSED_WORLD.header);
}

#[test]
fn test_parse_properties() {
  assert_debug_snapshot!(PARSED_WORLD.properties);
}

#[test]
fn test_parse_status() {
  assert_debug_snapshot!(PARSED_WORLD.status);
}
