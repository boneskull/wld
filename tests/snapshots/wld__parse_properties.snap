---
source: tests/wld.rs
expression: PARSED_WORLD.properties
---
Properties {
    tile_frame_importances: VariableTBitVec(
        BitVec<Lsb0, u8> [
            0b00011100,
            0b00111111,
            0b11111100,
            0b10111101,
            0b01111000,
            0b00100000,
            0b00100001,
            0b00000100,
            0b00000001,
            0b11100111,
            0b01111111,
            0b11111111,
            0b11111111,
            0b11100010,
            0b01100000,
            0b00000110,
            0b11001111,
            0b11110111,
            0b10000100,
            0b00000000,
            0b00000100,
            0b00011110,
            0b00100000,
            0b11110000,
            0b00000000,
            0b01000001,
            0b01101001,
            0b11111000,
            0b00011001,
            0b01011111,
            0b11111111,
            0b00000010,
            0b00000000,
            0b00000111,
            0b00011111,
            0b11110111,
            0b11111111,
            0b11111111,
            0b11111110,
            0b00101111,
            0b10011000,
            0b00000011,
            0b01110000,
            0b00000100,
            0b00111011,
            0b11111000,
            0b00001111,
            0b11101000,
            0b00111111,
            0b11110000,
            0b00000110,
            0b00111110,
            0b00011001,
            0b11011100,
            0b00011111,
            0b11111100,
            0b00001111,
            0b11000111,
            0b11111100,
        ],
    ),
    name: TString(
        "Foon",
    ),
    generator: GeneratorInfo {
        seed: TString(
            "1451234789",
        ),
        version: 833223655425,
    },
    uuid: TUuid(
        d578e106-3827-f648-a224-254c06ca78cb,
    ),
    id: 1468463142,
    bounds: Rect {
        left: 0,
        right: 67200,
        top: 0,
        bottom: 19200,
    },
    height: 1200,
    width: 4200,
    is_expert: False,
    created_on: 9860045932737703464,
    style: WorldStyle {
        moon: 1,
        trees: QuadrantStyle {
            x1: 3072,
            x2: 4200,
            x3: 4200,
            far_left: 4,
            near_left: 5,
            near_right: 0,
            far_right: 0,
        },
        moss: QuadrantStyle {
            x1: 1210,
            x2: 4200,
            x3: 4200,
            far_left: 1,
            near_left: 0,
            near_right: 3,
            far_right: 0,
        },
        underground_snow: 3,
        underground_jungle: 0,
        hell: 0,
    },
    spawn_point: Point {
        x: 2098,
        y: 229,
    },
    underground_level: 300.0,
    cavern_level: 528.0,
    current_time: 34958.0,
    is_daytime: True,
    moon_phase: 1,
    is_blood_moon: False,
    is_eclipse: False,
    dungeon_point: Point {
        x: 3426,
        y: 211,
    },
    evil_type: Corruption,
}
